### ISSUE TITLE
[FR] JSON API

### ISSUE THREAD

**Issue by God-damnit-all** [globalwordnet/english-wordnet#1110]
I'm requesting the implementation of a JSON API for https://en-word.net

I was recommended to create this issue by @jmccrae in https://github.com/globalwordnet/english-wordnet/issues/1012#issuecomment-2399260504:

> > Do you plan on making an API available?
>
> There are no immediate plans to make a well-documented API available, but you can use the linked data interface at https://en-word.net/. 
>
> That site has an undocumented, internal API that could easily be turned into a JSON API if required (see https://github.com/jmccrae/wordnet-angular/blob/master/src/main.rs). Can I suggest you make an issue and if it gets enough support we can do it.

**Comment by jmccrae** [comment]
Will be implemented as part of #1031

**Comment by God-damnit-all** [comment]
@jmccrae I don't see anything about an API anywhere. Is it just the JSON files?

**Comment by jmccrae** [comment]
Right, this issue is for the JSON API. It is underway and will be available with the 2026 release.

**Comment by jmccrae** [comment]
API is here https://en-word.net/api/docs

**Comment by 1313ou** [comment]
The [OEWN TK](https://github.com/oewntk/project) now offers JSON API support in the form of :
- a [server](https://github.com/oewntk/server) based on the KTOR platform
- a [client stub](https://github.com/oewntk/client) that can seamlessly deserialize model objects, can do its own processing of raw JSON if needed. But of course any JSON client will do, including _curl_

The server 
- is avaliable on [Maven Central](https://central.sonatype.com/artifact/io.github.oewntk/oewn-server)
- reads YAML and JSON models
- offers several JSON APIs (flat / hierachical) as per different [schemas](https://github.com/oewntk/server/tree/master/src/main/resources/schema)
- performs regex searches
- can provide schemas over the wire
- offers robust performance (based on KTOR)

See also the project's [json schemas](https://github.com/oewntk/json-validate/tree/master/src/main/resources)
Now [released](https://github.com/oewntk/server/releases) as an uber-jar.

**Comment by God-damnit-all** [comment]
> The [OEWN TK](https://github.com/oewntk/project?rgh-link-date=2026-09-16T09%3A52%3A32.000Z) now offers JSON API support in the form of :
> 
> * a [server](https://github.com/oewntk/server?rgh-link-date=2026-09-16T09%3A52%3A32.000Z) based the KTOR platform
> * a [client stub](https://github.com/oewntk/client?rgh-link-date=2026-09-16T09%3A52%3A32.000Z) that can seamlessly deserialize model objects, can do its own processing of raw JSON if needed. But of course any JSON client will do, including _curl_
> 
> The server
> 
> * is avaliable on [Maven Central](https://central.sonatype.com/artifact/io.github.oewntk/oewn-server)
> * reads YAML and JSON models
> * offers several JSON APIs (flat / hierachical) as per different [schemas](https://github.com/oewntk/server/tree/master/src/main/resources/schema?rgh-link-date=2026-09-16T09%3A52%3A32.000Z)
> * performs regex searches
> * can provide schemas over the wire
> * offers robust performance (based on KTOR)
> 
> See also the project's [json schemas](https://github.com/oewntk/json-validate/tree/master/src/main/resources?rgh-link-date=2026-09-16T09%3A52%3A32.000Z)

Thanks awesome, thank you!
